src/main/java/com/powsybl/openloadflow/network/impl/LfLoadImpl.java

```diff
@@ -97,7 +97,7 @@ void updateState(double diffLoadTargetP, boolean loadPowerFactorConstant) {
         for (int i = 0; i < loadsRefs.size(); i++) {
             Load load = loadsRefs.get(i).get();
             double updatedP0 = (load.getP0() / PerUnit.SB + diffLoadTargetP * participationFactors[i]) * PerUnit.SB;
-            double updatedQ0 = loadPowerFactorConstant ? getPowerFactor(load) * updatedP0 : load.getQ0();
+            double updatedQ0 = loadPowerFactorConstant && updatedP0 != 0.0 ? getPowerFactor(load) * updatedP0 : load.getQ0();
```

Comment:
Maybe it is more consistent to have :
double updatedQ0 = (load.getQ0() / PerUnit.SB + diffLoadTargetP * getPowerFactor(load) * participationFactors[i]) * PerUnit.SB;

Reply:
yes okay